api/python/quilt3/workflows/config-1.schema.json

```diff
@@ -0,0 +1,77 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://quiltdata.com/workflows/config/1",
+  "type": "object",
+  "required": [
+    "version",
+    "workflows"
+  ],
+  "additionalProperties": false,
+  "properties": {
+    "version": {
+      "const": 1
+    },
+    "is_workflow_required": {
+      "type": "boolean",
+      "description": "Whether package can be pushed with unspecified workflow.",
+      "default": false
+    },
+    "default_workflow": {
+      "type": "string",
+      "description": "The workflow that will be used if no workflow provided by user."
+    },
+    "workflows": {
+      "type": "object",
+      "minProperties": 1,
+      "propertyNames": {
+        "pattern": "^[A-Za-z_-][A-Za-z0-9_-]*$",
+        "maxLength": 64
+      },
```

Comment:
@akarve 
I don't have clear reasons for these constraints, I added them just in case we'd want to rework this somehow in future: like putting workflows in separate files or something like this.
Does that make sense?

Reply:
It feels overly restrictive since any python or even JS access will quote key names? I'n not against this but it might frustrate users. We had this with quilt packages names: they were very restrictive and users kept bumping into this limit.

Reply:
These constraints inspired by [GH workflow config](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_id). docker-compose has [similar ones](https://github.com/docker/compose/blob/675c9674e1701cc22763bab2bcd429ac5b82e102/compose/config/compose_spec.json#L18). The general idea is that in the future making it less restrictive is easier than making it more restrictive, because the latter is a breaking change. Also users have to quote it CLI.